### PR TITLE
- Added StartModalService with function to get title from start event…

### PR DESCRIPTION
**Pull request by PeterK-Ritense**
- Added StartModalService with function to get title from start event to be used in start modal component.

- Modified start modal component to get modal title from start event. If no start event name found it will fall back to original 'start - ${process-name}' start form title format.

### CODE REVIEW

File: projects/valtimo/dossier/src/lib/services/start-modal.service.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  providedIn: 'root'
+})
+export class StartModalService {
+  private valtimoEndpointUri: string;
+
+  constructor(
+    private http: HttpClient,
+    private configService: ConfigService
+  ) {
+    this.valtimoEndpointUri = configService.config.valtimoApi.endpointUri;
```
Comment (mbritense) [inline projects/valtimo/dossier/src/lib/services/start-modal.service.ts]: Why is this here if it's never used

File: projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts [hunk 2]
```diff
@@ -75,6 +81,9 @@ export class DossierProcessStartModalComponent implements OnInit {
     this.processLinkId = null;
     this.formDefinition = null;
     this.formFlowInstanceId = null;
+    this.processService.getProcessDefinitionXml(this.processDefinitionId).subscribe((result) => {
```
Comment (mbritense) [inline projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts]: I'd like to see this call only executed if the feature flag is enable, now it always happens.

File: projects/valtimo/dossier/src/lib/services/start-modal.service.ts (hunk 1 above)
Comment (PeterK-Ritense) [reply projects/valtimo/dossier/src/lib/services/start-modal.service.ts]: Ah yeah, not needed anymore, removed

File: projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts (hunk 2 above)
Comment (PeterK-Ritense) [reply projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts]: Added if statement to check for the feature toggle

File: projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts [hunk 3]
```diff
@@ -75,6 +81,11 @@ export class DossierProcessStartModalComponent implements OnInit {
     this.processLinkId = null;
     this.formDefinition = null;
     this.formFlowInstanceId = null;
+    if (this.configService.config.featureToggles?.useStartEventNameAsStartFormTitle) {
```
Comment (mbritense) [inline projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts]: can you make this a property you set in the component so you dont need to duplicate it on line 119?

File: projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts [hunk 4]
```diff
@@ -45,6 +47,8 @@ export class DossierProcessStartModalComponent implements OnInit {
   public processDefinitionId: string;
   public documentDefinitionName: string;
   public processName: string;
+  public startEventName: string;
+  public useStartEventNameAsStartFormTitle: boolean;
```
Comment (mbritense) [inline projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts]: Is this used in component code? Otherwise `private _useStartEventNameAsStartFormTitle!: boolean;` Same goes for other class prop

**Review by mbritense — APPROVED** [review]
